zend_parse_arg_test.go

```diff
@@ -0,0 +1,291 @@
+	// sample object <nil>
+	// <nil> unsupported type : gophplib.Sample2
+	// <nil> unsupported type : []int
+	// <nil> unsupported type : []string
+	// <nil> unsupported type : *os.File
+	// <nil> unsupported type : gophplib.CustomType
+}
+
+func TestZend_parse_arg(t *testing.T) {
+	file, osErr := os.Open("README.md")
+	if osErr != nil {
+		panic(osErr)
+	}
+
+	testCase := []struct {
+		testName string
+		input    any
+		expected any
+	}{
+		{
+			testName: "Plain string",
+			input:    "Hello world",
+			expected: "Hello world",
+		},
+		{
+			testName: "Empty string",
+			input:    "",
+			expected: "",
+		},
+		{
+			testName: "StringWithSpecialChars",
+			input:    "Line1\nLine2\tTab",
+			expected: "Line1\nLine2\tTab",
+		},
+		{
+			testName: "Int",
+			input:    123,
+			expected: "123",
+		},
+		{
+			testName: "Int64",
+			input:    9223372036854775807,
+			expected: "9223372036854775807",
+		},
+		{
+			testName: "Negative int",
+			input:    -123,
+			expected: "-123",
+		},
+		{
+			testName: "Float64",
+			input:    123.456,
+			expected: "123.456",
+		},
+		{
+			testName: "Float64 exceeds 14 digits",
+			input:    123.456789012345678,
+			expected: "123.45678901235",
+		},
+		{
+			testName: "Exponent",
+			input:    10.1234567e10,
+			expected: "101234567000",
+		},
+		{
+			testName: "Special Float - NaN",
+			input:    math.NaN(),
+			expected: "NAN",
+		},
+		{
+			testName: "Special float - positive infinity",
+			input:    math.Inf(1),
+			expected: "INF",
+		}, {
+			testName: "Special float - negative infinity",
+			input:    math.Inf(-1),
+			expected: "-INF",
+		},
+		{
+			testName: "Exponent",
+			input:    10.1234567e10,
+			expected: "101234567000",
+		},
+		{
+			testName: "Negative float64",
+			input:    -123.456,
+			expected: "-123.456",
+		},
+		{
+			testName: "Zero float64",
+			input:    0.0,
+			expected: "0",
+		},
+		{
+			testName: "True",
+			input:    true,
+			expected: "1",
+		},
+		{
+			testName: "False",
+			input:    false,
+			expected: "",
+		},
+		{
+			testName: "Nil",
+			input:    nil,
+			expected: "",
+		},
+		{
+			testName: "Object with toString function",
+			input:    Sample{},
+			expected: "sample object",
+		},
+		{
+			testName: "Object without toString function",
+			input:    Sample2{},
+			expected: nil,
+		},
+		{
+			testName: "Int array",
+			input:    []int{1, 2, 3},
+			expected: nil,
+		},
+		{
+			testName: "String array",
+			input:    []string{"hello", "world"},
+			expected: nil,
+		},
+		{
+			testName: "NestedArray",
+			input:    []interface{}{[]interface{}{1, 2}, []interface{}{"a", "b"}},
+			expected: nil,
+		},
+		{
+			testName: "Resource",
+			input:    file,
+			expected: nil,
+		},
+		{
+			testName: "Custom type",
+			input:    CustomType{"Hello world"},
+			expected: nil,
+		},
+	}
+	for _, tc := range testCase {
+		t.Run(tc.testName, func(t *testing.T) {
+			result, err := Zend_parse_arg(tc.input)
+			if err != nil {
+				assert.Equal(t, result, nil)
```

Comment:
Since golang does not support test-only-dependencies or dev-only-dependencies, all dependencies used for testing are treated equally with ordinary dependencies which is not desirable since it’ll bloat library users’ dependency list. So I’m trying to avoid any test-only-dependencies.

For me `assert.Equal` looks quite obviously replaceable with ordinary if and `t.Errorf`. Let’s replace it and remove testify dependency.

Reply:
I've removed the dependency on testify.